pkg/cmd/server/apis/config/types.go

```diff
@@ -642,6 +642,7 @@ type MasterNetworkConfig struct {
 	// For security reasons, you should ensure that this range does not overlap with the CIDRs reserved for external ips,
 	// nodes, pods, or services.
 	IngressIPNetworkCIDR string
+	VXLANPort            *uint32
```

Comment:
I'm not sure if we need it to be optional here; I don't think MasterNetworkConfig is exposed externally

Reply:
@danwinship I changed this in response to @pravisankar's [link](https://github.com/openshift/origin/pull/20024#discussion_r196227037) and @smarterclayton's [link](https://github.com/openshift/api/pull/57#discussion_r201200256) comments.

Reply:
Clayton's comment was only about public API objects. There's no compatibility concern with objects that are not part of the API. I don't think this object is part of the API.

Reply:
not sure what you mean by that. this is exposed in config files, which is part of API (externally visible, supported across releases)

Reply:
If you're referring to the `+optional` bits, then you are correct... that isn't important for config files. As long as this properly responds to loading an old config file and getting a zero value, that's sufficient.

Reply:
`0` isn't a valid value for a port, so if we read in the config file and get 0, we can assume that that means it wasn't specified, and replace it with the default value. (Though I guess that means that if they explicitly say `vxlanPort: 0`, we'll treat that as "use the default" rather than "fail with an error"...)